### PR TITLE
[IE] support different modes for Interpolate

### PR DESCRIPTION
**Pull request by DariaMityagina**
MDK : #14750

### CODE REVIEW

File: inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp [hunk 1]
```diff
@@ -126,6 +139,8 @@ TEST_P(myriadResampleLayerTests_smoke, Resample) {
 
     std::map<std::string, std::string> params;
     params["antialias"] = std::to_string((int)antialias);
+    params["coordTransMode"] = std::to_string((int)coordTransMode);
```
Comment (andrejsokolov) [inline inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp:142]: What typical values can `coordTransMode` have? Are they numbers converted to strings? Or just names like "asymmetric"?

File: inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp [hunk 2]
```diff
@@ -61,12 +63,21 @@ void refResample(const Blob::Ptr src, Blob::Ptr dst, int antialias) {
...
             {
-                float ix = ox*fx + fx / 2.0f - 0.5f;
+                float ix = ox*fx + fx / 2.0f - 0.5f; // half pixel
                 float iy = oy*fy + fy / 2.0f - 0.5f;
 
-                int ix_r = (int)(round(ix));
+                int ix_r = (int)(round(ix)); // round prefer ceil
                 int iy_r = (int)(round(iy));
 
+                if (coordTransMode == 2) { // asymmetric
```
Comment (andrejsokolov) [inline inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp:72]: Is it any chance that we can reuse `InterpolateCoordTransMode` here to avoid using `int`?

Comment (DariaMityagina) [reply inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp:72]: I guess, yes. Will do.

Comment (DariaMityagina) [reply inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp]: done.

**Comment by DariaMityagina** [comment]
@andrejsokolov , could you look one more time, please?

File: inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp [hunk 3]
```diff
@@ -126,6 +154,8 @@ TEST_P(myriadResampleLayerTests_smoke, Resample) {
 
     std::map<std::string, std::string> params;
     params["antialias"] = std::to_string((int)antialias);
+    params["coordTransMode"] = std::to_string(coordTransMode);
```
Comment (andrejsokolov) [inline inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp:157]: What values this parameter can have? Integer or text like "half_pixel"?

Comment (DariaMityagina) [reply inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp:157]: Here, in case of tests, Integer.

Comment (andrejsokolov) [reply inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp:157]: but what does parser of layer get? Does it take text representation of parameters?

Comment (DariaMityagina) [reply inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp:157]: Parser gets text representation, but in tests it is easier to use it's integer value (according to the enum declaration), IMO

Comment (andrejsokolov) [reply inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp:157]: when you fulfill map `params`, "cordTransMoe" = "some_int_value". Parser gets such line, not text representation. The test create parameters for the layer, and these parameters are integers converted to string. Or I am mistaken?

Comment (DariaMityagina) [reply inference-engine/tests_deprecated/functional/vpu/common/layers/myriad_layers_resample_test.hpp:157]: Yes, you are correct.
Found where I was mistaken. After fulfilling map params, I do not process these values in parseResample method, but assign these parameters the default value, that is why it acts normal and why i didn't really looked at the parameters map before -> gets text and works fine. Interpolate stage calls addResampleNearestStage with correct parameters, but Resample - with the default ones.

File: inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp [hunk 4]
```diff
@@ -18,10 +18,14 @@ const std::vector<InferenceEngine::Precision> netPrecisions = {
```
Comment (andrejsokolov) [inline inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp:18]: Did you add new modes for testing to the new test system?

Comment (DariaMityagina) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp:18]: I tried new modes, but there still is some difference between ref and res. It is now definitely much closer to the reference Interpolate with these modes. But there still can be a few failures.

Comment (andrejsokolov) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp:18]: it looks strange, because new modes are supported now. What exactly tests failed? With what combinations of modes for them?

**Comment by AndrewBakalinIntel** [comment]
@DariaMityagina is it still "WIP"? May we remove this prefix?

**Comment by DariaMityagina** [comment]
> @DariaMityagina is it still "WIP"? May we remove this prefix?

sure

File: inference-engine/cmake/vpu_dependencies.cmake [hunk 5]
```diff
@@ -19,8 +19,8 @@ set(VPU_SUPPORTED_FIRMWARES usb-ma2x8x pcie-ma248x)
 # Default packages
 #
 
-set(FIRMWARE_PACKAGE_VERSION 1452)
-set(VPU_CLC_MA2X8X_VERSION "movi-cltools-20.09.2")
+set(FIRMWARE_PACKAGE_VERSION 1462)
+set(VPU_CLC_MA2X8X_VERSION "movi-cltools-20.09.1")
```
Comment (AndrewBakalinIntel) [inline inference-engine/cmake/vpu_dependencies.cmake:23]: Is this change needed?

File: inference-engine/src/vpu/graph_transformer/include/vpu/model/stage.hpp [hunk 6]
```diff
@@ -270,6 +270,21 @@ VPU_DECLARE_ENUM(BroadcastMode,
 // StageDataInfo
 //
 
+VPU_DECLARE_ENUM(InterpolateCoordTransMode,
+    half_pixel = 0,
```
Comment (AndrewBakalinIntel) [inline inference-engine/src/vpu/graph_transformer/include/vpu/model/stage.hpp:274]: Could you please use PascalCase for these values?

File: inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp [hunk 7]
```diff
@@ -51,12 +51,31 @@ void FrontEnd::parseInterpolate(const Model& model, const ie::CNNLayerPtr& _laye
             if (cmp(interpolateMode, "nearest")) {
                 // current "Resample" supports the following "Interpolate" modes only:
                 // coordinate_transformation_mode = half_pixel; nearest_mode = round_prefer_ceil;
+                // coordinate_transformation_mode = asymmetric; nearest_mode = floor;
                 // other "Interpolate" modes are translated to the default ones
                 const auto antialias = _layer->GetParamAsBool("antialias", false);
+                const auto coordinate_transformation_mode = _layer->GetParamAsString("coordinate_transformation_mode", "half_pixel");
```
Comment (AndrewBakalinIntel) [inline inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp:57]: Please use camelCase for the variable names.

File: inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp [hunk 8]
```diff
@@ -51,12 +51,31 @@ void FrontEnd::parseInterpolate(const Model& model, const ie::CNNLayerPtr& _laye
...
+                const auto nearest_mode = _layer->GetParamAsString("nearest_mode", "round_prefer_floor");
+                InterpolateCoordTransMode coordinateTransformationMode = InterpolateCoordTransMode::half_pixel;
+                InterpolateNearestMode nearestMode = InterpolateNearestMode::round_prefer_ceil;
+
+                if (cmp(coordinate_transformation_mode, "asymmetric")) {
+                    coordinateTransformationMode = InterpolateCoordTransMode::asymmetric;
```
Comment (AndrewBakalinIntel) [inline inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp:63]: Are there any other `coordinateTransformationMode`'s or `nearestMode`'s? If yes, I think we should also throw on unsupported ones.

Comment (andrejsokolov) [reply inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp:63]: No, we should do not throw an exception. We should use default mode.

Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp:63]: Ok, got it.

File: inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp (hunk 4 above)
Comment (DariaMityagina) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp:18]: ok, I will restore those tests after corrections for review and place the results here.

Comment (DariaMityagina) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp:18]: well, there are no failures any longer) I checked the calculations one more time(there was my mistake), modified the formula a bit. Tests pass now.

**Comment by DariaMityagina** [comment]
@AndrewBakalinIntel @andrejsokolov could you take one more look, please?

File: inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp (hunk 4 above)
Comment (AndrewBakalinIntel) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp:18]: So can we now remove the added tests from deprecated one, in order to avoid duplication?

Comment (DariaMityagina) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp:18]: but if I revert changes in the deprecated tests it'll cause failures, right? because I changed stages and tests should also be modified.

Comment (AndrewBakalinIntel) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp:18]: Actually if the tests for new test system cover the cases from deprecated one, we can simply remove deprecated tests to avoid tests duplication.

Comment (DariaMityagina) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp:18]: done

**Review by AndrewBakalinIntel — APPROVED** [review]

**Comment by DariaMityagina** [comment]
@andrejsokolov do you have any other comments?

**Review by andrejsokolov — APPROVED** [review]

**Comment by andrejsokolov** [comment]
If validation with the latest changes is passed, I guess it is ok.
